compiler/modules/nback/init.bal

```diff
@@ -242,21 +266,146 @@ function createMappingDescType(t:Context tc, t:SemType semType) returns llvm:Str
     return llvm:structType([LLVM_TID, "i32", LLVM_MEMBER_TYPE, fillerDescPtrType, llvm:arrayType(LLVM_MEMBER_TYPE, mat.names.length())]);
 }
 
-function createMappingDescInit(InitModuleContext cx, int tid, t:SemType semType) returns llvm:ConstValue {
+function createMappingDescInit(InitModuleContext cx, llvm:Builder builder, int tid, t:SemType semType) returns llvm:ConstValue {
     t:MappingAtomicType mat = <t:MappingAtomicType>t:mappingAtomicType(cx.tc, semType);
     llvm:ConstValue[] llFields = from var ty in mat.types let var tyInner = t:cellInner(ty) select getMemberType(cx, tyInner);
     t:SemType rest = t:cellInnerVal(mat.rest);
     return cx.llContext().constStruct([
         constTid(cx, tid),
         constI32(cx, llFields.length()),
         getMemberType(cx, rest),
-        getFillerDesc(cx, rest),
+        getFillerDesc(cx, builder, rest),
         cx.llContext().constArray(LLVM_MEMBER_TYPE, llFields)
     ]);
 }
 
+function createFunctionDescType(t:Context tc, t:SemType semType) returns llvm:StructType {
+    t:FunctionAtomicType atomic = <t:FunctionAtomicType>t:functionAtomicType(tc, semType);
+    t:FunctionSignature signature = t:functionSignature(tc, atomic);
+    return llvm:structType([LLVM_TID,
+                            llvm:pointerType(llUniformFunctionType),
+                            LLVM_MEMBER_TYPE,
+                            LLVM_MEMBER_TYPE,
+                            LLVM_INT,
+                            llvm:arrayType(LLVM_MEMBER_TYPE, requiredParamCount(signature))]);
+}
+
+function createFunctionDescInit(InitModuleContext cx, llvm:Builder builder, int tid, t:SemType semType, string symbol) returns llvm:ConstValue {
+    t:FunctionAtomicType atomic = <t:FunctionAtomicType>t:functionAtomicType(cx.tc, semType);
+    t:FunctionSignature signature = t:functionSignature(cx.tc, atomic);
+    llvm:Function uniformFunc = createUniformFunction(cx, signature, builder, symbol + "_uniform");
+    llvm:ConstValue returnTy = getMemberType(cx, signature.returnType);
+    t:SemType? restParamType = signature.restParamType;
+    llvm:ConstValue restParamTy = restParamType == () ? getMemberType(cx, t:NEVER) : getMemberType(cx, restParamType);
+    int nRequiredParams = requiredParamCount(signature);
+    llvm:ConstValue paramTys = cx.llContext().constArray(LLVM_MEMBER_TYPE,
+                                                         from int i in 0 ..< nRequiredParams select getMemberType(cx, signature.paramTypes[i]));
+    return cx.llContext().constStruct([constTid(cx, tid), uniformFunc, returnTy, restParamTy,
+                                       constInt(cx, nRequiredParams), paramTys]);
+}
+
+function createUniformFunction(InitModuleContext cx, t:FunctionSignature signature, llvm:Builder builder, string symbol) returns llvm:Function {
```

Comment:
In other places in `nback` we have `llvm:Builder` followed by `context` in the arg list. Let's follow that.

Reply:
It seems the convention we have followed in `init.bal` is `InitModuleContext` followed by `llvm:Builder`. Changed to match that